app/Http/Controllers/MessageGroupController.php

```diff
@@ -70,4 +81,39 @@ public function store(Request $request)
 
         return new CreatedResponse($messageGroup);
     }
+
+    /**
+     * @param MessageGroup $group
+     * @return Response|UnauthorizedResponse
+     */
+    public function show(MessageGroup $group)
+    {
+        if (!$this->session->player->message_groups->contains($group)) {
+            return new UnauthorizedResponse();
+        }
+
+        return new Response($group->load('message_group_members.player'));
+    }
+
+    /**
+     * @param MessageGroup $group
+     * @param Request $request
+     * @return UnauthorizedResponse
+     */
+    public function update(MessageGroup $group, Request $request)
+    {
+        // It is nonsensical to update a MessageGroup as it has no real properties on its own.
+        return new UnauthorizedResponse();
+    }
+
+    /**
+     * @param MessageGroup $group
+     * @return UnauthorizedResponse
+     */
+    public function destroy(MessageGroup $group)
+    {
+        // Perhaps an admin should be allowed to delete a chat group?
+        // But it makes no sense for any of the group members to delete the whole chat.
+        return new UnauthorizedResponse();
```

Comment:
Maybe this could allow a player to leave one of their chats if someone is adding them to groups they don't want to be in

Reply:
Wouldn't this get very messy? As soon as people can leave/be added to chats, you don't have a unique room for each subset of players. The same subset of players could have multiple groups. What's to stop two players from having 5 chats with each other XD

Reply:
I think if a group of players want to create multiple groups that they're all in, that's mess they made for themselves; it doesn't really matter to us.

Note though that this is not what this endpoint is about; this controller is about _entire_ groups, not about single player addition/removal within a group.

Currently, there are no endpoints that add or remove players from a chat group: all its participants are decided upon creation and are fixed thereafter.

Reply:
Yes, I think that would be the best design in game, and also I would presume easiest from a coding perspective?

Reply:
Ah, my bad, didn't realize this was deleting the whole group. Yeah, I don't think we should be able to delete a group. I believe that would delete all messages with it? But yeah the more I think about it, just leave it. If they create a mess of group chats that is on them. No use putting more effort into this than we have to

Reply:
In the hypothetical scenario that we would allow deletion of a group (say some Subs admin could, not an indiviaul player), then yes, the messages would get deleted as well due to the foreign key having an `on delete cascade` setting. All foreign keys are like that, actually. For example if an entire player would get deleted, then he's also automatically removed from all rooms, groups, all his messages are deleted, and so on. In other words, there can be no "dangling" data.

I think we can resolve this discussion now (but I will leave that up to you guys), but the question whether a player should be able to leave a chat group he was put in remains and perhaps we should design a (separate) endpoint, just like the leave room endpoint, to leave a group.